Extension/src/LanguageServer/client.ts

```diff
@@ -144,6 +145,33 @@ function publishIntelliSenseDiagnostics(params: PublishIntelliSenseDiagnosticsPa
     clients.timeTelemetryCollector.setUpdateRangeTime(realUri);
 }
 
+function publishRefactorDiagnostics(params: PublishRefactorDiagnosticsParams): void {
+    if (!diagnosticsCollectionRefactor) {
+        diagnosticsCollectionRefactor = vscode.languages.createDiagnosticCollection(CppSourceStr);
+    }
+
+    const newDiagnostics: vscode.Diagnostic[] = [];
+    params.diagnostics.forEach((d) => {
+        const message: string = getLocalizedString(d.localizeStringParams);
+        const diagnostic: vscode.Diagnostic = new vscode.Diagnostic(makeVscodeRange(d.range), message, d.severity);
+        diagnostic.code = d.code;
+        diagnostic.source = CppSourceStr;
+        if (d.relatedInformation) {
+            diagnostic.relatedInformation = [];
+            for (const info of d.relatedInformation) {
+                diagnostic.relatedInformation.push(new vscode.DiagnosticRelatedInformation(makeVscodeLocation(info.location), info.message));
+            }
+        }
+
+        newDiagnostics.push(diagnostic);
+    });
+
+    const fileUri: vscode.Uri = vscode.Uri.parse(params.uri);
+    diagnosticsCollectionRefactor.set(fileUri, newDiagnostics);
+
+    clients.timeTelemetryCollector.setUpdateRangeTime(fileUri);
```

Comment:
I believe this was used for telemetry in `publishIntelliSenseDiagnotics`, to determine when the first set of squiggles is received, and probably shouldn't be duplicated here.